### PR TITLE
ClojureScript support

### PR DESCRIPTION
**Pull request by jeluard**
I did some work around ClojureScript support. Is that something you would consider a pull request for?

### CODE REVIEW

**Comment by jeluard** [comment]
I started migrating [pathetic](https://github.com/davidsantiago/pathetic) [here](https://github.com/davidsantiago/pathetic/pull/3) as part of this effort.

**Comment by jeluard** [comment]
Most tests now pass (but 2 as I dont know how to translate them). Also I had to bump the visibility of 2 methods as `'#` is not valid in ClojureScript.

Otherwise everything looks good!

**Comment by jeluard** [comment]
@cemerick , is this something you would consider merging?

File: src/cemerick/url.cljx [hunk 1]
```diff
@@ -24,17 +38,27 @@
...
+
+#+cljs
+(defn split
+  ([string delim] (string/split string (re-pattern delim)))
+  ([string delim limit] (string/split string (re-pattern delim) limit)))
+
 (defn split-param [param]
   (->
-   (.split param "=")
+   (split param "=")
```
Comment (cemerick) [inline src/cemerick/url.cljx]: Might as well just use `string/split` here all the time, and eliminate the two `split` impls above.

File: src/cemerick/url.cljx [hunk 2]
```diff
@@ -69,6 +93,41 @@
...
+        [user pass] (split (or (.getUserInfo url) "") ":" 2)]
+    (URL. (.getScheme url)
+          (and (seq user) user)
+          (and (seq pass) pass)
+          (.getDomain url)
+          (translate-default (.getPort url) nil -1)
+          (pathetic/normalize (.getPath url))
+          (query->map (translate-default (.getQuery url) "" nil))
+          (translate-default (.getFragment url) "" nil))))
+
```
Comment (cemerick) [inline src/cemerick/url.cljx:120]: `url*` would be a more obvious name, I think.  Also, these should probably be private?

**Comment by cemerick** [comment]
Would be happy to merge after a couple of tweaks, I added some notes.  Questions:
- "traduce"?  Looks like one test can be re-enabled widely (we can ditch Clojure 1.2.0 at this point).
- Could you squash the commits down?
- Do you know when cljs-compatible pathetic will be non-SNAPSHOT?

Thanks for working on this!

File: src/cemerick/url.cljx (hunk 1 above)
Comment (jeluard) [inline src/cemerick/url.cljx]: Right I will simplify that.

File: src/cemerick/url.cljx (hunk 2 above)
Comment (jeluard) [inline src/cemerick/url.cljx:120]: Good point!

**Comment by jeluard** [comment]
- "traduce"
  Ahah sorry I meant translate.
- Could you squash the commits down?
  Ok I will do that!
- Do you know when cljs-compatible pathetic will be non-SNAPSHOT?
  I believe we are waiting for the `cljx` release to switch to the new syntax.

**Comment by cemerick** [comment]
re: cljx release, ah-ha.  Coming _very_ soon, I suspect, squashed the last known bug this morning.

**Comment by jeluard** [comment]
I made the changes related to your comments and squashed all commits.
Now waiting for `cljx` and `pathetic` releases.

**Comment by jeluard** [comment]
Pathetic 0.5.0 has been released, travis build is now successful!

**Comment by jeluard** [comment]
Can you see some issue left?

**Comment by cemerick** [comment]
No, just haven't gotten back to it yet.  Soon :-)

**Comment by cemerick** [comment]
Thanks!

**Pull request merged**